### PR TITLE
`sage.rings`: Modularization fixes for singular

### PR DESCRIPTION
**Pull request by mkoeppe**
Removing the use of a Singular interpreter instance as a default value of parameters

- Cherry-picked from #35095.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/rings/polynomial/polynomial_singular_interface.py [hunk 1]
```diff
@@ -74,6 +74,9 @@ def _do_singular_init_(singular, base_ring, char, _vars, order):
          //        block   2 : ordering C,
          None)
     """
+    if singular is None:
+        from sage.interfaces.singular import singular
+
     make_ring = lambda s: singular.ring(s, _vars, order=order)
```
Comment (kwankyu) [inline src/sage/rings/polynomial/polynomial_singular_interface.py]: This does not look right. The caller should provide `singular`.

File: src/sage/rings/polynomial/polynomial_singular_interface.py [hunk 2]
```diff
@@ -333,8 +336,10 @@ def _singular_(self, singular=singular):
         """
         try:
             R = self.__singular
-            if not (R.parent() is singular):
+            if singular is not None and R.parent() is not singular:
                 raise ValueError
+            elif singular is None:
+                from sage.interfaces.singular import singular
             R._check_valid()
```
Comment (kwankyu) [inline src/sage/rings/polynomial/polynomial_singular_interface.py:336]: Perhaps line 339 and line 341 should be reversed. Also note that in line 45, `singular` is already imported...

File: src/sage/rings/polynomial/polynomial_singular_interface.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/rings/polynomial/polynomial_singular_interface.py]: Indeed. I've removed it in ee8875e89cb

File: src/sage/rings/polynomial/polynomial_singular_interface.py (hunk 2 above)
Comment (mkoeppe) [reply src/sage/rings/polynomial/polynomial_singular_interface.py:336]: Done in f90914eaad9

> note that in line 45, `singular` is already imported...

Removed now in ece9c930230

**Review by kwankyu — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**